src/modclub_assets/src/utils/gtm.ts

```diff
@@ -0,0 +1,18 @@
+type GTMEvent = {
+  event: string;
+  [key: string]: any;
+};
+
+const GTMManager = {
+  _pushToDataLayer: (event: GTMEvent): void => {
+    if (window && window.dataLayer) {
+      window.dataLayer.push(event);
+    }
+  },
+
+  trackEvent: (eventName: string, eventData: object): void => {
```

Comment:
Is it possible to specify a more precise type for eventData?

Reply:
an object can always be different, it is difficult to predict what fields will be in an object